### ISSUE TITLE
Redefine Codensity and Yoneda in terms of Ran

### ISSUE THREAD

**Issue by joneshf** [FormationAI/dhall-bhat#30]
Creating an issue for visibility, and so I don't forget to do this :sweat_smile:.

---

As discussed in slack, there's a formulation of `Codensity` and `Yoneda` in terms of Kan extensions. In particular, given the right Kan extension:

```Dhall
-- Ran
  λ(f : Type → Type)
→ λ(g : Type → Type)
→ λ(a : Type)
→ ∀(b : Type) → (a → f b) → g b
```

We can define `Codensity`:

```Dhall
-- Codensity
λ(m : Type → Type) → Ran m m
```

We can also define `Yoneda`:

```Dhall
Ran Id
```

Does that seem about right? More than happy to submit a PR for this.

**Comment by FintanH** [comment]
@joneshf Yup! That looks legit 😄

**Comment by FintanH** [comment]
When you're happy to submit a PR can you run `make lint` and `make compile` to ensure that everything is clean and compiles :) we'll eventually get CI to confirm that this is done, but unfortunately it isn't done yet.

**Comment by joneshf** [comment]
Want a PR for that as well?

**Comment by FintanH** [comment]
We may be adding CI on our side. Need to ping our boy in Formation to check the status on that. Thanks for the offer 😄

**Comment by FintanH** [comment]
Closed via #31
